test/e2e/page-objects/pages/home/asset-list.ts

```diff
@@ -235,8 +243,18 @@ class AssetListPage {
       this.tokenImportSelectNetwork(chainId),
     );
     await this.driver.fill(this.tokenAddressInput, tokenAddress);
-    await this.driver.fill(this.tokenSymbolInput, symbol);
+
+    if (symbol) {
+      // do not fill the form until the button is disabled, because there's a form re-render which can clear the input field causing flakiness
+      await this.driver.waitForSelector(this.importTokensNextButton, {
+        state: 'disabled',
+      });
+      await this.driver.fill(this.tokenSymbolInput, symbol);
+    }
+
+    await this.driver.waitForSelector(this.tokenDecimals);
     await this.driver.clickElement(this.importTokensNextButton);
+    await this.driver.waitForSelector(this.tokenConfirmListItem);
```

Comment:
this is to avoid a race condition where the token is not loaded in the modal, so if we click confirm while that happens, we get no token imported